server/core/src/main/java/org/infinispan/server/core/MagicByteDetector.java

```diff
@@ -27,6 +27,11 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) {
       }
       byte b = in.getByte(in.readerIndex());
       if (b == magicByte) {
+         if (!server.isDefaultCacheRunning()) {
```

Comment:
What is the goal here? This will just read one more byte and then decode will be called again later after more bytes are written and the magic byte won't match again and it will just remove the detector without installing anything.. if I understand correctly. What is the goal here?